File: email/src/main/scala/br/com/santos/william/email/message/MessageConfig.scala
package br.com.santos.william.email.message

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import org.springframework.amqp.core.{Exchange, ExchangeBuilder}
import org.springframework.amqp.support.converter.{Jackson2JsonMessageConverter, MessageConverter}
import org.springframework.beans.factory.annotation.{Autowired, Value}
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component

@Component
@Autowired
class MessageConfig(@Value("${message.amqp.exchange:EMAIL_EXCHANGE}") val exchange: String) {

  @Bean
  def declareExchange(): Exchange = ExchangeBuilder.directExchange(exchange)
    .durable(true).build();

  @Bean
  def declareMessageConverter(): MessageConverter = new Jackson2JsonMessageConverter(objectMapper())

  @Bean
  def objectMapper(): ObjectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
}

File: auth/src/main/kotlin/br/com/santos/william/auth/message/MessageConfig.kt
package br.com.santos.william.auth.message

import org.springframework.amqp.core.Exchange
import org.springframework.amqp.core.ExchangeBuilder
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.amqp.support.converter.MessageConverter
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class MessageConfig @Autowired constructor(
        @Value("\${message.amqp.exchange.user:USER_EXCHANGE}") val exchange: String
) {

    @Bean
    fun declareExchange(): Exchange = ExchangeBuilder.directExchange(exchange)
            .durable(true).build()

    @Bean
    fun declareMessageConverter(): MessageConverter = Jackson2JsonMessageConverter()
}